### PR TITLE
API/OP (center_loss, fluid.one_hot, prroi_pool, roi_pool, ctc_greed_decoder) error message enhancement

### PR DESCRIPTION
**Pull request by LDOUBLEV**
error message enhanced

- center_loss
- fluid.one_hot 
- prroi_pool 
- roi_pool 
- ctc_greed_decoder

### CODE REVIEW

File: paddle/fluid/operators/roi_pool_op.cc [hunk 1]
```diff
@@ -26,39 +26,45 @@ class ROIPoolOp : public framework::OperatorWithKernel {
...
-                   "The format of input tensor is NCHW.");
-    PADDLE_ENFORCE(rois_dims.size() == 2,
-                   "ROIs should be a 2-D LoDTensor of shape (num_rois, 4)"
-                   "given as [[x1, y1, x2, y2], ...].");
-    PADDLE_ENFORCE(rois_dims[1] == kROISize,
-                   "ROIs should be a 2-D LoDTensor of shape (num_rois, 4)"
-                   "given as [[x1, y1, x2, y2], ...].");
+    PADDLE_ENFORCE_EQ(input_dims.size(), 4,
+                      platform::errors::InvalidArgument(
+                          "The format of input tensor is NCHW."));
```
Comment (chenwhql) [inline paddle/fluid/operators/roi_pool_op.cc]: tell users error firstly?

File: paddle/fluid/operators/roi_pool_op.cc [hunk 2]
```diff
@@ -26,39 +26,45 @@ class ROIPoolOp : public framework::OperatorWithKernel {
...
+    PADDLE_ENFORCE_EQ(
+        rois_dims.size(), 2,
+        platform::errors::InvalidArgument(
+            "ROIs should be a 2-D LoDTensor of shape (num_rois, 4)"
```
Comment (chenwhql) [inline paddle/fluid/operators/roi_pool_op.cc]: LoDTensor of shape? -> with shape is better?

File: paddle/fluid/operators/roi_pool_op.cc [hunk 3]
```diff
@@ -26,39 +26,45 @@ class ROIPoolOp : public framework::OperatorWithKernel {
...
+            "given as [[x1, y1, x2, y2], ...]."));
+    PADDLE_ENFORCE_EQ(
+        rois_dims[1], kROISize,
+        platform::errors::InvalidArgument(
+            "ROIs should be a 2-D LoDTensor of shape (num_rois, 4)"
```
Comment (chenwhql) [inline paddle/fluid/operators/roi_pool_op.cc]: same above

File: paddle/fluid/operators/roi_pool_op.cc [hunk 4]
```diff
@@ -26,39 +26,45 @@ class ROIPoolOp : public framework::OperatorWithKernel {
...
+            "given as [[x1, y1, x2, y2], ...]."));
 
     int pooled_height = ctx->Attrs().Get<int>("pooled_height");
     int pooled_width = ctx->Attrs().Get<int>("pooled_width");
     float spatial_scale = ctx->Attrs().Get<float>("spatial_scale");
 
     PADDLE_ENFORCE_GT(pooled_height, 0,
-                      "The pooled output height must greater than 0");
+                      platform::errors::OutOfRange(
+                          "The pooled output height must greater than 0"));
```
Comment (chenwhql) [inline paddle/fluid/operators/roi_pool_op.cc]: must be greater than?

File: paddle/fluid/operators/roi_pool_op.cc [hunk 5]
```diff
@@ -26,39 +26,45 @@ class ROIPoolOp : public framework::OperatorWithKernel {
...
     PADDLE_ENFORCE_GT(pooled_width, 0,
-                      "The pooled output width must greater than 0");
-    PADDLE_ENFORCE_GT(spatial_scale, 0.0f,
-                      "The spatial scale must greater than 0");
+                      platform::errors::OutOfRange(
+                          "The pooled output width must greater than 0"));
```
Comment (chenwhql) [inline paddle/fluid/operators/roi_pool_op.cc]: same above

File: paddle/fluid/operators/roi_pool_op.cc [hunk 6]
```diff
@@ -26,39 +26,45 @@ class ROIPoolOp : public framework::OperatorWithKernel {
...
+    PADDLE_ENFORCE_GT(
+        spatial_scale, 0.0f,
+        platform::errors::OutOfRange("The spatial scale must greater than 0"));
```
Comment (chenwhql) [inline paddle/fluid/operators/roi_pool_op.cc]: same above

**Review by qingqing01 — DISMISSED** [review]
LGTM.

**Review by chenwhql — COMMENTED** [review]
关于size和height相关的对比，麻烦后面说明一下实际值，例如
```
PADDLE_ENFORCE_GT(pooled_height, 0,
                      platform::errors::OutOfRange(
                          "The pooled output height must greater than 0, but received height is %d", pooled_height));
```

File: paddle/fluid/operators/roi_pool_op.cc (hunk 1 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/roi_pool_op.cc]: Done

File: paddle/fluid/operators/roi_pool_op.cc (hunk 2 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/roi_pool_op.cc]: Done

File: paddle/fluid/operators/roi_pool_op.cc (hunk 3 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/roi_pool_op.cc]: Done

File: paddle/fluid/operators/roi_pool_op.cc (hunk 4 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/roi_pool_op.cc]: Done

File: paddle/fluid/operators/roi_pool_op.cc (hunk 5 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/roi_pool_op.cc]: Done

File: paddle/fluid/operators/roi_pool_op.cc (hunk 6 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/roi_pool_op.cc]: Done

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**